src/tools/tidy/src/style.rs

```diff
@@ -62,10 +63,40 @@ const ANNOTATIONS_TO_IGNORE: &[&str] = &[
     "// normalize-stderr-test",
 ];
 
-// Intentionally written in decimal rather than hex
-const PROBLEMATIC_CONSTS: &[u32] = &[
-    184594741, 2880289470, 2881141438, 2965027518, 2976579765, 3203381950, 3405691582, 3405697037,
-    3735927486, 3735932941, 4027431614, 4276992702,
+fn generate_problems<'a>(consts: &'a[u32], letter_digit: &'a HashMap<char, char>) -> impl Iterator<Item = u32> + 'a {
+    consts.into_iter().flat_map(move |const_value| {
+        let mut problem = format!("{:X}", const_value);
+        for (key, value) in letter_digit {
+            problem = problem.replace(&value.to_string(), &key.to_string());
+        }
+        let indexes: Vec<usize> = problem.chars().enumerate().filter_map(|(index, c)| {
+            if letter_digit.contains_key(&c) {
+                Some(index)
+            }
+            else {
+                None
+            }
+        }).collect();
+        (0..1 << indexes.len()).map(move |i| {
+            let result = problem.chars().enumerate().map(|(index, c)| {
+                if let Some(pos) = indexes.iter().position(|&x| x == index) {
+                    if (i >> pos) & 1 == 1 {
+                        letter_digit[&c]
+                    } else {
+                        c
+                    }
+                } else {
+                    c
+                }
+            }).collect::<String>();
+            u32::from_str_radix(&result, 0x10).unwrap()
+        })
+    })
+}
+
+const ROOT_PROBLEMATIC_CONSTS: &[u32] = &[
+        184594741, 2880289470, 2881141438, 2965027518, 2976579765, 3203381950, 3405691582, 340569703,
```

Comment:
You lost a digit from the end of this line.
```suggestion
        184594741, 2880289470, 2881141438, 2965027518, 2976579765, 3203381950, 3405691582, 3405697037,
```